cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -477,7 +477,10 @@ Future<Void> maybeRemoveOldClusterCaCertificates() {
                     // only if all Kafka related components pods are updated to the new cluster CA cert generation,
                     // there is the possibility that we should remove the older cluster CA from the Secret and stores
                     for (Pod pod : pods) {
-                        int podClusterCaCertGeneration = Integer.parseInt(pod.getMetadata().getAnnotations().get(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION));
+                        int podClusterCaCertGeneration =
+                                pod.getMetadata().getAnnotations() != null && pod.getMetadata().getAnnotations().containsKey(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION) ?
+                                        Integer.parseInt(pod.getMetadata().getAnnotations().get(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION)) :
+                                        clusterCaCertGeneration;
```

Comment:
What about using something like this:
```suggestion
                        int podClusterCaCertGeneration = Annotations.intAnnotation(pod, Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, clusterCaCertGeneration);
```

Reply:
Absolutely! I forgot we have that `Annotations` utility class.